doc/usage/index.rst

```diff
@@ -25,3 +25,64 @@ Once the daemon is running, you need to choose which front-end's CLI to use:
 - :doc:`bfcli`: ``bpfilter``-specific CLI, developed as part of the project. ``bfcli`` supports new ``bpfilter`` features before other CLIs as it's used for development. It allows for a more flexible rule definition: you can use a combination of filters and hooks that might not be possible with other CLIs. However, it doesn't support ``nftables`` or ``iptables`` rules format.
 - :doc:`nftables`: requires a custom version of the ``nft`` binary with ``bpfilter`` support (see below), and support for new ``bpfilter`` features is usually a bit delayed.
 - :doc:`iptables`: similar to ``nftables``, however ``iptables`` has been deprecated globally in favor of ``nftables``.
+
+Example Usage
+-------------
+
+.. note::
+
+	This is only to be used as an example and a more interactive test to familiarize you with ``bpfilter``, more in-depth information can be found throughout the docs.
+
+Initialize ``bpfilter`` daemon
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: bash
+
+	> cd path/to/bpfilter/repo/
+	> sudo ./build/output/bpfilter --transient --verbose=debug --no-iptables
+
+
+Load ``bfcli`` filter(s)
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+While the ``bpfilter`` daemon runs, now we will open up a separate window to use ``bfcli``
+
+.. note::
+
+	``bfcli`` is just one of the ways you can communicate with the ``bpfilter`` daemon along with ``iptables`` and ``nftables``
+
+.. code-block:: bash
+
+	> cd path/to/bpfilter/
+	> sudo ./build/output/bfcli --str "chain BF_HOOK_NF_LOCAL_IN policy ACCEPT rule ip4.saddr eq 192.168.1.1 ACCEPT"
+
+The above command is just to make sure that ``bfcli`` is able to communicate with ``bpfilter``, but it's still worth working through it. This command is telling ``bpfilter`` to check incoming traffic from the ``BF_HOOK_NF_LOCAL_IN`` hook location and accept those connections. Then the command asks ``bpfilter`` to check to see if the incoming IP address is equal to ``192.168.1.1`` (yourself) and to accept the connection if that is true. When you run the commands you should see output from ``bpfilter`` registering that a filter has been loaded.
```

Comment:
This part is confusing and inaccurate:
- "``bpfilter`` to check incoming traffic": it sends a request to `bpfilter` to create a new filter. `bpfilter` itself doesn't process the traffic
- "accept those connections": "connections" when it comes to packet filtering have a very specific meaning (e.g. TCP connection), `bpfilter`'s programs works at the packet level
- "asks ``bpfilter`` to check": `bpfilter` won't check anything but generate a program that does
- "``192.168.1.1`` (yourself)": that is very specific to one's network.

I would prefer something simpler and more straightforward such as:

The command above will send a request to the `bpfilter` daemon to create a chain and attach it to the `BF_HOOK_NF_LOCAL_IN` hook, located after the packet has been routed in the kernel network stack. A single rule is defined, which will accept the packet if its source IPv4 address is `192.168.1.1`. If the rule doesn't match the packet, the chain's policy is applied and the packet is accepted. In its current shape, the chain will always accept incoming packets.